enhancements/20221228-various-storage-class-support.md

```diff
@@ -0,0 +1,114 @@
+# Various StorageClass support of Harvester CSI Driver
+
+## Summary
+
+Harvester supports the user-defined StorageClass after v1.1.0. Because of that, Harvester CSI Driver should also let users use the custom StorageClass mentioned above.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/3052
+
+## Motivation
+
+### Goals
+
+The user using rancher integration with Harvester Cluster (Called Guest Cluster) could create the volume with the different StorageClass associated with StorageClass on the Host Harvester Cluster.
+
+## Proposal
+
+We will implement the mechanism to get the StorageClass parameter from the Guest Cluster and try to create it on the Host Harvester Cluster.
+
+### User Stories
+
+#### General use cases
+
+Generally, users use the default StorageClass(`harvester`). The default StorageClass has three replicas means higher reliability. Users do not do any extra configuration. Create volume with the default StorageClass(`harvester`), then use.
+
+#### More specific goals
+
+Sometimes users will have different purposes for storage volume. Like `data archiving` or `Storage tiering`. For the above requirements, users can change the data locality to HDD. Users can complete it by creating a new StorageClass and configuring its data locality to HDD. Also, for the `performance` purpose. Users can reduce the number of replicas (remember, the default replica number is three). Reducing the replica number from 3 to 2 would increase the volume performance, but it lost some data reliability. So, users can create a new StorageClass and config its replica number to 2 and make the storage volume with this StorageClass to get better performance. 
+
+### User Experience In Detail
+
+- Harvester CSI Driver no longer hardcodes the `longhorn` StorageClass name when creating volume from the Guest Cluster. Furthermore, The default StorageClass is associated with the `Default` StorageClass on the host Harvester Cluster.
```

Comment:
Not specific to this line, but if `host Harvester Cluster` and `Host Harvester Cluster` are both referring to the same thing it'd be better to use the same form throughout the doc to avoid ambiguity that they're different things.

Reply:
OK, I will check again on this term! Thanks for the reminder!